### PR TITLE
Pass SVGs as fds

### LINKED ISSUE #521: Pass SVG files as fds, not as path strings
When we're packaging the client in a sandbox (i.e. flatpak, see https://github.com/libratbag/piper/pull/235) we don't have access to the host file system and we can't just load the SVG ourselves. The two options we have is either open the file system to gain access or, simpler and better, passing the SVG as a file descriptor instead of a path.

### PR DESCRIPTION
**Pull request by whot**
This is on top of #522 
Fixes #521 and is the foundation for https://github.com/libratbag/piper/issues/245

This is an API break so we'll need to push a release out and then we can update piper to rely on that release.

### CODE REVIEW

**Comment by whot** [comment]
fwiw, I have a commit pending to piper at last to check the ratbagd version on build. So this somewhat mitigates this and I don't know whether there's anyone else who cares at this point.

**Comment by Hjdskes** [comment]
Ignoring the API discussion, the code looks good to me.

**Comment by bentiss** [comment]
> and I'm planning to release both soon after this is merged anyway, so presumably anyone who cares will update both.

Your call then. I still have my concerns, but if you think we won't have too many "piper broke", we can break the API.

I do wonder also if you should not update `ratbagc.py` as you are editing `ratbagctl.in` (hint: I do not wonder that much, I know you have to).

**Comment by whot** [comment]
> I do wonder also if you should not update ratbagc.py as you are editing ratbagctl.in (hint: I do not wonder that much, I know you have to).

`ratbagc.py` is a direct libratbag interface, right? the fd handling here is a ratbagd component, not exposed by libratbag which remains the same as before. `ratbagctl` needs the update because it uses ratbagd but that's already part of the patch

File: ratbagd/ratbagd-device.c [hunk 1]
```diff
@@ -167,14 +167,14 @@ static int ratbagd_device_get_theme_svg(sd_bus_message *m,
 	const char *theme;
 	const char *svg;
 	const char *datadir;
+	int fd = -1;
```
[the remark this answers is not in the document]
Comment (whot) [reply ratbagd/ratbagd-device.c]: ack, fixed in the next version, thanks

**Comment by whot** [comment]
Pushed, changes to previous version:
* use `_cleanup_close_` for the fd
* make the API `GetSvgFd()` and leave the old one in-place

**Pull request merged**